src/sequence_manager.cpp

```diff
@@ -61,25 +61,28 @@ bool SequenceManager::sequenceExists(const uint64_t sequenceId) const {
 
 Status SequenceManager::removeTimedOutSequences() {
     std::unique_lock<std::mutex> sequenceManagerLock(mutex);
-
+    SPDLOG_LOGGER_DEBUG(sequence_manager_logger, "Sequence watcher thread for model {} version {} Manager lock acquired", modelName, modelVersion);
     for (auto it = sequences.begin(); it != sequences.end();) {
         Sequence& sequence = it->second;
         // Non blocking try to get mutex
         std::unique_lock<std::mutex> sequenceLock(sequence.getMutex(), std::try_to_lock);
         if (!sequence.isTerminated() && sequenceLock.owns_lock()) {
+            SPDLOG_LOGGER_DEBUG(sequence_manager_logger, "Sequence watcher thread for model {} version {} Sequence lock acquired Id: {}", modelName, modelVersion, sequence.getId());
             sequenceLock.unlock();
+            SPDLOG_LOGGER_DEBUG(sequence_manager_logger, "Sequence watcher thread for model {} version {} Sequence unlocked Id: {}", modelName, modelVersion, sequence.getId());
```

Comment:
I think all this logging about locking and unlocking should be removed before merge